### DISCUSSION TITLE
Add tray option and allow to run valent in background without opening gui

### DISCUSSION THREAD

**Discussion by isshin1** [andyholmes/valent#594] [Feature Request]
2 separate but related feature requests, hopefully they can be added 

Ist will be adding a tray icon for valent  which can display all options as a drop down menu , including battery status, sending files and whatnot 

another is that on running valent , GUI is opened, which is a bit inconvenient. if will be nice to have a --run-in-background or --tray flag to launch the binary

**Comment by andyholmes** [comment]
> Ist will be adding a tray icon for valent which can display all options as a drop down menu , including battery status, sending files and whatnot

This sounds like #431, which I guess needs AppIndicator ported to GTK4.

> another is that on running valent , GUI is opened, which is a bit inconvenient. if will be nice to have a --run-in-background or --tray flag to launch the binary

Activating the launcher will open the GUI, since otherwise there wouldn't really be a way to open it. Valent should auto-start with the session though, without opening the GUI.

The GNOME Shell extension can start/stop the service without opening the GUI, but that's only because it knows the special D-Bus incantations.

**Comment by isshin1** [comment]
> Valent should auto-start with the session though, without opening the GUI.

how does this work ? i dont see a systemctl service for it. Especially on non gnome environments

**Reply by andyholmes** [reply]
Valent has an [XDG Autostart](https://specifications.freedesktop.org/autostart-spec/autostart-spec-latest.html) file that gets installed ([`ca.andyholmes.Valent.desktop`](https://github.com/andyholmes/valent/blob/main/data/ca.andyholmes.Valent-autostart.desktop.in.in)).

If running in Flatpak, it should be installed by the XDG portal in `~/.config/autostart` I believe. When installed on the host, it should end up in `/etc/xdg/autostart`. Maybe your system doesn't have support for the [Background Portal](https://flatpak.github.io/xdg-desktop-portal/docs/#gdbus-org.freedesktop.portal.Background)?

**Reply by isshin1** [reply]
yea that was the issue , sway doesnt support this , had to manually add 
```
valent --gapplication-service
```
as a command to run on session startup as in desktop file. thanks
